architecture/adr-list/adr-003-cli-tools.md

```diff
@@ -0,0 +1,125 @@
+
+1. **Cosmos-based CLI:** Most likely route for Cosmos projects for their node application. Most existing Cosmos node validators will be familiar with this method of managing their node. 
+2. **VDR CLI**: Traditionally, a lot of SSI networks have used [Hyperledger Indy](https://github.com/hyperledger/indy-node) and therefore the Indy CLI tool for managing and interacting with the ledger. This has now been renamed to [Verifiable Data Registry \(VDR\) Tools CLI](https://gitlab.com/evernym/verity/vdr-tools) and is the tool that most existing SSI node operators \("stewards"\) would be familiar with.
+
+Ideally, the `cheqd-node` project would provide a consistent set of CLI tools rather than two separate tools with varying feature sets between them.
+
+This ADR will focus on the CLI tool architecture choice for `cheqd-node`.
+
+## Context
+
+### Assumptions / Considerations
+
+#### **Likelihood of introducing bugs or security vulnerabilities**
+
+1. Any CLI tool architecture chosen should not increase the likelihood of introducing bugs, security vulnerabilities, or design pattern deviations from upstream Cosmos SDK.
+2. Actions that are carried out on ledger through a CLI tool in `cheqd-node` now include token functionality as well as identity functionality. E.g., if a DID gets compromised, there could be mechanisms to recover or signal that fact to issuers/verifiers. If tokens or the staking balance of node operators get compromised, this may potentially have more severe consequences for them.
+
+#### User / Node Operator preferences
+
+1. Would node operators want a single CLI to manage everything?
+   1. This might be the case with node operators from an SSI / digital identity background, or node operators familiar with Hyperledger Indy CLI / VDR Tools CLI.
+   2. A “single CLI” could be a single tool as far as the user sees, but actually consist of multiple modules beneath it in how it’s implemented.
+2. Would node operators be okay with having two separate CLIs?
+   1. One for Cosmos-ledger functions, and one for identity-specific functions.
+   2. Unlike existing Hyperledger Indy networks, it is anticipated that some of the node operators on the cheqd network will have experience running Cosmos validator nodes. For this group, having to learn a new “single” CLI tool could cause a steeper learning curve and a worse user experience than what they have now.
+   3. Node operators may want one/separate CLIs for security and operational reasons, i.e., for a separation of concerns in terms of functionality.
+
+### Existing CLI architecture overview
+
+Currently, there are two separate CLI tools being used by the development team to interact with the ledger: Cosmos-based CLI and VDR CLI.
+
+![Current Cosmos CLI and VDR tools CLI architecture](https://lh3.googleusercontent.com/cMdfEe19vqDVaRJ0kP97KGCUHauEpnh2TV1OhmvGqOFqqIkhWXGkdKxONDLjW2rnU83k9yelFWK_jhsqQoF57tNf8ChrPeIZsiLys3LKVT_QKG9Gk7Mir4ChbCeiUKs2V7l7jE8d=s0)
```

Comment:
The image link here is to a Google Doc cache link that may not be consistently available or shown to specific readers. Could you please save the image separately, and it add it in? Can be done via Gitbook, as long as you're in the adr/cli-tools branch.

Reply:
Oh.. Didn't expect gitbook to do it. Sure.